cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java

```diff
@@ -668,11 +668,25 @@ public static int entries(HttpServletRequest request) {
     return parameterString == null ? Integer.MAX_VALUE : Integer.parseInt(request.getParameter(parameterString));
   }
 
+  /**
+   * @param values Integer values
+   * @return A set of negative integer values contained in the given set.
+   */
+  private static Set<Integer> getNegatives(Set<Integer> values) {
+    return values.stream().filter(v -> v < 0).collect(Collectors.toCollection(() -> new HashSet<>(values.size())));
```

Comment:
Just curious, why using `Collectors.toCollection(() -> new HashSet<>(values.size()))` instead of `Collectors.toSet()` here? 
Is it because it preallocate the capacity of the set?

Reply:
Yes, this provides a minor memory optimization if the given set is small -- which is the expectation in its current use-cases.